File: java/src/test/java/problems1_100/MergeKSortedListsTest.java
package problems1_100;

import concepts.SinglyListNode;
import java.util.Arrays;
import org.junit.Test;
import testing.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class MergeKSortedListsTest extends SampledTest {
    @Test
    public void test() {
        for (MergeKSortedLists solution : MergeKSortedLists.values()) {
            for (MergeKSortedListsSample sample : getSamples(MergeKSortedListsSample[].class)) {
                assertWithMessage(sample.getMessage())
                    .that(
                        solution.mergeKLists(
                            Arrays
                                .stream(sample.input.lists)
                                .map(SinglyListNode::of)
                                .toList()
                                .toArray(new SinglyListNode[sample.input.lists.length])
                        )
                    ).isEqualTo(SinglyListNode.of(sample.output));
            }
        }
    }
}

File: kotlin/src/test/kotlin/problems1_100/MergeKSortedListsTest.kt
package problems1_100

import com.google.common.truth.Truth.assertWithMessage
import concepts.SinglyListNode
import sample.SampledTest
import sample.component1
import sample.component2
import sample.component3
import kotlin.test.Test

class MergeKSortedListsTest : SampledTest() {
    @Test
    fun test() =
        MergeKSortedLists.entries.forEach { solution ->
            getSamples(Array<MergeKSortedListsSample>::class.java)
                .forEach { (input, output, message) ->
                    assertWithMessage(message)
                        .that(
                            solution.mergeKLists(
                                input.lists
                                    .map { SinglyListNode.of(*it) }
                                    .toTypedArray<SinglyListNode?>(),
                            ),
                        ).isEqualTo(SinglyListNode.of(*output))
                }
        }
}
